### PR TITLE
net: sockets: socket_service: change callback arg to `struct net_socket_service_event`

### PR DESCRIPTION
**Pull request by maass-hamburg**
This removes the remains from when there was a async mode for the socket service. 
Mainly it changes the callback argument from `struct k_work *work` to `struct net_socket_service_event`.

### CODE REVIEW

File: subsys/net/lib/shell/sockets.c [hunk 1]
```diff
@@ -123,10 +123,7 @@ static void walk_socket_services(const struct net_socket_service_desc *svc,
 	snprintk(owner, sizeof(owner), "<unknown>");
```
Comment (jukkar) [inline subsys/net/lib/shell/sockets.c:123]: The commit subject can be optimized a bit, no need to mention so much stuff so
```
net: shell: sockets: Don't show service type
```
is enough.

File: include/zephyr/net/socket_service.h [hunk 2]
```diff
@@ -81,8 +81,6 @@ struct net_socket_service_desc {
 #define __z_net_socket_svc_get_idx(_svc_id) __z_net_socket_service_idx_##_svc_id
```
Comment (jukkar) [inline include/zephyr/net/socket_service.h:89]: The commit subject `net: sockets: socket_service: ...` has redundant information, could be simplified to
`net: socket: service: Optimize code`

File: include/zephyr/net/socket_service.h [hunk 3]
```diff
@@ -32,15 +32,23 @@
 extern "C" {
```
Comment (jukkar) [inline include/zephyr/net/socket_service.h:32]: Also please change `net: sockets: socket_service: remove k_work related code` to
`net: sockets: service: Remove k_work related code`

**Review by jukkar — COMMENTED** [review]
Overall LGTM, just minor nits about commit subjects

File: subsys/net/lib/shell/sockets.c [hunk 4]
```diff
@@ -123,10 +123,7 @@ static void walk_socket_services(const struct net_socket_service_desc *svc,
...
 #endif
 
-	PR("%32s  %-6s  %-5d %s\n",
```
Comment (jukkar) [inline subsys/net/lib/shell/sockets.c:126]: You forgot to change the commit subject of the shell changes.

Comment (maass-hamburg) [reply subsys/net/lib/shell/sockets.c:126]: Oops

**Review by jukkar — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]

**Comment by maass-hamburg** [comment]
forgot to change it for the sample

**Review by pdgendt — DISMISSED** [review]

**Review by rlubos — COMMENTED** [review]
Looks good, but it seems there's still some leftovers in LLMNR code.

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**